socorro/cron/jobs/verify_processed.py

```diff
@@ -0,0 +1,151 @@
+from psycopg2 import IntegrityError
+
+from socorro.cron.base import BaseCronApp
+from socorro.cron.mixins import as_backfill_cron_app
+from socorro.lib.dbutil import execute_no_results
+from socorro.lib.transaction import transaction_context
+
+
+metrics = markus.get_metrics('crontabber.verifyprocessed')
+
+
+def check_crashids(entropy, boto_conn, bucket_name, date):
+    """Checks crash ids for a given entropy and date."""
+    bucket = boto_conn.get_bucket(bucket_name)
+
+    raw_crash_prefix_template = 'v2/raw_crash/%s/%s/'
+    processed_crash_template = 'v1/processed_crash/%s'
+    raw_crash_key_prefix = raw_crash_prefix_template % (entropy, date)
+
+    missing = []
+    for key_instance in bucket.list(raw_crash_key_prefix):
+        raw_crash_key = key_instance.key
+        crash_id = raw_crash_key.split('/')[-1]
+
+        processed_crash_key = bucket.get_key(processed_crash_template % crash_id)
+        if processed_crash_key is None:
+            missing.append(crash_id)
+
+    return missing
+
+
+@as_backfill_cron_app
+class VerifyProcessedCronApp(BaseCronApp):
+    app_name = 'verifyprocessed'
+    app_description = 'verifies that incoming crash reports were processed'
+    app_version = '1.0'
+
+    required_config = Namespace()
+    required_config.add_option(
+        'crashstorage_class',
+        default='socorro.external.boto.crashstorage.BotoS3CrashStorage',
+        doc='S3 crash storage class',
+        from_string_converter=class_converter,
+        reference_value_from='resource.boto'
+    )
+    required_config.add_option(
+        'database_class',
+        default='socorro.external.postgresql.connection_context.ConnectionContext',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql'
+    )
+    required_config.add_option(
+        'date',
+        default='backfill',
+        doc=(
+            'The YYYYMMDD date. By default, this will use the backfill date from '
+            'the db.'
+        )
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.database = self.config.database_class(self.config)
+        self.crashstorage = self.config.crashstorage_class(self.config)
+
+    def get_entropy(self):
+        """Generate all entropy combinations."""
+        chars = '0123456789abcdef'
+        for x in chars:
+            for y in chars:
+                for z in chars:
+                    yield x + y + z
+
+    def find_missing_multiprocessing(self, date):
+        connection_source = self.crashstorage.connection_source
+        bucket_name = connection_source.config.bucket_name
+        boto_conn = connection_source._connect()
+
+        num_workers = 5
+        chunksize = 1
+
+        check_crashids_for_date = partial(
+            check_crashids,
+            boto_conn=boto_conn,
+            bucket_name=bucket_name,
+            date=date
+        )
+
+        pool = multiprocessing.Pool(num_workers)
```

Comment:
This uses multiprocessing to split the task into smaller ones that run in multiple processes getting around Python's GIL and making better use of waiting around for network requests to happen. This can probably be tweaked--I picked 5 and 1 because that seemed fine.